internal/services/cognitive/cognitive_account_connection_resource_test.go

```diff
@@ -0,0 +1,491 @@
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("api_key", "metadata"),
+	})
+}
+
+func TestAccCognitiveAccountConnection_oauth2(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cognitive_account_connection", "test")
+	r := CognitiveAccountConnectionTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.oauth2(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("oauth2", "metadata"),
+	})
+}
+
+func TestAccCognitiveAccountConnection_customKeys(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cognitive_account_connection", "test")
+	r := CognitiveAccountConnectionTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.customKeys(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("custom_keys", "metadata"),
+	})
+}
+
+func TestAccCognitiveAccountConnection_AAD(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cognitive_account_connection", "test")
+	r := CognitiveAccountConnectionTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.AAD(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("metadata"),
+	})
+}
+
+func TestAccCognitiveAccountConnection_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cognitive_account_connection", "test")
+	r := CognitiveAccountConnectionTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("api_key", "metadata"),
+		{
+			Config: r.apiKey(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("api_key", "metadata"),
+	})
+}
+
+func TestAccCognitiveAccountConnection_updateStorageBlob(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cognitive_account_connection", "test")
+	r := CognitiveAccountConnectionTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.oauth2(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("oauth2", "metadata"),
+		{
+			Config: r.oauth2Updated(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("oauth2", "metadata"),
+		{
+			Config: r.AAD(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("metadata"),
+	})
+}
+
+func TestAccCognitiveAccountConnection_authTypeValidation(t *testing.T) {
```

Comment:
Let's either remove all of these or include the other two as well.

Reply:
removed